envwarden

```diff
@@ -48,6 +48,20 @@ while [[ $# > 0 ]]; do
     shift
 done
 
+which jq > /dev/null 2>&1
+
+if [ $? != 0 ]; then
+  (>&2 echo "jq command not found" )
+  exit 1
+fi
+
+JQ_VERSION=`jq --version 2>&1 | cut -f3 -d" "`
+
+if (( $(echo "$JQ_VERSION < 1.6" |bc -l) )); then
```

Comment:
This will fail if jq releases version 1.6.1 for example... I'm not sure, but I imagine that jq uses semantic versioning? and comparing them is a bit difficult unfortunately... I didn't think about it initially

Maybe something like this can help? https://gist.github.com/ObserverOfTime/761d89b39d77b5d3fce2da9564a78b45 (but it's a fairly big chunk of code for something relatively small, so not sure.

You previously mentioned 1.3, but here you're checking for 1.6. Do you know which distros / versions have < 1.6? I'm curious how many people might have an outdated jq

Reply:
This is the error you get when using a jq version prior to 1.6 us used, not very helpful;

error: Invalid character
.[].fields[]? | select(.name != null) | select(.value != null) | .name, .value
            ^
1 compile error
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at WriteWrap.afterWrite [as oncomplete] (net.js:832:14)
Emitted 'error' event at:
    at onwriteError (_stream_writable.js:431:12)
    at onwrite (_stream_writable.js:456:5)
    at _destroy (internal/streams/destroy.js:40:7)
    at Socket.stdout._destroy (internal/process/stdio.js:24:7)
    at Socket.destroy (internal/streams/destroy.js:32:8)
    at WriteWrap.afterWrite [as oncomplete] (net.js:834:10)

Reply:
> This will fail if jq releases version 1.6.1 for example... I'm not sure, but I imagine that jq uses semantic versioning? and comparing them is a bit difficult unfortunately... I didn't think about it initially

Maybe an alternative?
`
MINIMAL_VER="1.6"

for CURR_VER in 1.1 1.0.1 1.00.1 1.3 1.6 1.6.1 1.7.0.1; do
  if [ "$CURR_VER" != "$MINIMAL_VER" ]; then
    printf '%s\n%s' $CURR_VER $MINIMAL_VER | sort -C -V
    if [ $? == 0 ]; then
        echo $CURR_VER is lower than $MINIMAL_VER
    else
        echo $CURR_VER is higher than $MINIMAL_VER
    fi
  else
    echo $CURR_VER is equal to $MINIMAL_VER
  fi
done
`

Reply:
> You previously mentioned 1.3, but here you're checking for 1.6. Do you know which distros / versions have < 1.6? I'm curious how many people might have an outdated jq

Yes because I was using 1.3, so  I knew it needed to be newer than that.

From here I think to understand the []? syntax was introduced in version 1.6

stedolan/jq@fe29d3d

I know 1.3 is shipped with Centos 6 and Ubuntu 14 at least.